### PR TITLE
test: temporarily allow 10 CA restarts

### PR DESCRIPTION
**Pull request by mboersma**
**Reason for Change**:

Provides a temporary workaround for #1597 so we don't miss other failures that might be occurring in CI. cluster-autoscaler restarting several times on Linux but coming up `Ready` is still something to investigate, but the overall test is useful, so this disables testing for just that specific and logs it instead.

**Issue Fixed**:

Refs #1597

**Requirements**:

- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Notes**:

### CODE REVIEW

File: test/e2e/kubernetes/kubernetes_test.go [hunk 1]
```diff
@@ -520,7 +520,12 @@ var _ = Describe("Azure Container Cluster using the Kubernetes Orchestrator", fu
 			for _, currentPod := range pods.Pods {
 				log.Printf("Checking %s - ready: %t, restarts: %d", currentPod.Metadata.Name, currentPod.Status.ContainerStatuses[0].Ready, currentPod.Status.ContainerStatuses[0].RestartCount)
 				Expect(currentPod.Status.ContainerStatuses[0].Ready).To(BeTrue())
-				Expect(currentPod.Status.ContainerStatuses[0].RestartCount).To(BeNumerically("<", 5))
+				tooManyRestarts := 5
+				if strings.Contains(currentPod.Metadata.Name, "cluster-autoscaler") && !eng.HasWindowsAgents() {
+					log.Print("need to investigate cluster-autoscaler restarts on Linux!")
```
Comment (PatrickLang) [inline test/e2e/kubernetes/kubernetes_test.go]: nothing in kube-system runs on Windows, so I don't see any need for if eng.HasWindowsAgents(). I think whatever tests we run against deployments in kube-system should be the same regardless of which node OS are used.

TL;DR - I'm ok with 10 restarts everywhere

Comment (CecileRobertMichon) [reply test/e2e/kubernetes/kubernetes_test.go]: +1
